src/commands/CountryCommand.ts

```diff
@@ -0,0 +1,391 @@
+import { Command } from "../Command";
+import { CommandInteraction, MessageEmbed } from "discord.js";
+import {
+	inlineCode,
+	SlashCommandBuilder,
+	SlashCommandStringOption,
+	SlashCommandSubcommandsOnlyBuilder,
+} from "@discordjs/builders";
+import {
+	Country,
+	countryData,
+	CountryKey,
+	getCountryByName,
+	getFilteredCountryDataBy,
+	initializeCountryData,
+	sortCountryDataBy,
+	typeOfCountryProperty,
+} from "../util/countryDataManager";
+
+export let formatNumber: (n: number) => string;
+
+class CountryCommand extends Command {
+	constructor() {
+		super("country");
+		initializeCountryData();
+	}
+
+	async execute(interaction: CommandInteraction): Promise<void> {
+		formatNumber = (n: number) => {
+			return n.toLocaleString(interaction.locale);
+		};
+
+		if (!countryData) {
+			interaction.reply({ content: "Still initiliazing the data, try again later...", ephemeral: true });
+			return;
+		}
+
+		switch (interaction.options.getSubcommand()) {
+			case "overview": {
+				const country: Country | undefined = getCountryByName(interaction.options.getString("country") ?? "Bhutan");
```

Comment:
This fixes the typing errors without unnecessary default value which never gets used.
Same piece of code applies to the some of the other options done elsewere.
```suggestion
				const country: Country | undefined = getCountryByName(interaction.options.getString("country", true));
```